src/platform/src/Bridge/VertexAi/TokenOutputProcessor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\Platform\Bridge\VertexAi;
+
+use Symfony\AI\Agent\Output;
+use Symfony\AI\Agent\OutputProcessorInterface;
+use Symfony\AI\Platform\Result\Metadata\Metadata;
+use Symfony\AI\Platform\Result\StreamResult;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Junaid Farooq <ulislam.junaid125@gmail.com>
+ */
+final class TokenOutputProcessor implements OutputProcessorInterface
+{
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     */
+    public function processOutput(Output $output): void
+    {
+        $metadata = $output->result->getMetadata();
+
+        if ($output->result instanceof StreamResult) {
```

Comment:
Cf https://github.com/symfony/ai/pull/347#discussion_r2297300546 I'm dunno if StreamResult should be handled here...

Maybe it's better to have the same behavior everywhere `// Streams have to be handled manually as the tokens are part of the streamed chunks` and to open an issue first to discuss about it ?

Reply:
I left a comment there.